src/Synthesiser.cpp

```diff
@@ -1335,13 +1338,43 @@ void Synthesiser::emitCode(std::ostream& out, const RamStatement& stmt) {
                 }
                 out << ",";
             }
-            // extra 0 for provenance height annotation
+            // extra 0 for provenance height annotations
+            for (int i = 0; i < (int)(numberOfHeights - 1); i++) {
```

Comment:
Since you're just counting, use size_t and avoid the casting. You may need to count from 1 to numberOfHeights to avoid underflow, though

Reply:
Adapted in clean up commit. There is no risk of underflows when computing `numberOfHeights - 1` as there is always at least one height parameter.